test/integration/http_integration.cc

```diff
@@ -422,7 +422,6 @@ void HttpIntegrationTest::testDrainClose() {
                                                                {":authority", "host"}},
                                        *response_);
   response_->waitForEndStream();
-  codec_client_->waitForDisconnect();
```

Comment:
? Why this deletion?

Reply:
The actual disconnection is happening. but waitForDisconnect is returning false.. probably because of other dispatcher events...that is why removed it.
   [2018-03-20 00:30:19.289][3380685][debug][connection] source/common/network/connection_impl.cc:134] [C85] closing socket: 1
[2018-03-20 00:30:19.289][3380685][debug][client] source/common/http/codec_client.cc:81] [C85] disconnect. resetting 0 pending requests
[2018-03-20 00:30:19.289][3381124][trace][connection] source/common/network/connection_impl.cc:386] [C86] socket event: 1
[2018-03-20 00:30:19.289][3381124][trace][connection] source/common/network/connection_impl.cc:424] [C86] read ready
[2018-03-20 00:30:19.289][3381124][trace][connection] source/common/network/raw_buffer_socket.cc:25] [C86] read returns: 0
test/integration/http_integration.cc:142: Failure
Value of: disconnected_
  Actual: false
Expected: true

Reply:
I don't understand how your change could have effected this test? There is no idle timeout here? I'm guessing a previous test (one you added) is not cleaning up properly?

Reply:
@mattklein123 found the problem. Some of the integration tests are using MockClusterInfo for creating http connection. Since now the MockClusterInfo returns the valid idle timeout value, idle timeout is happening and client is closing. The LocalClose event is generated which is not handled by the IntegrationCodecClient. I have increased the idle timeout value and reverted the waitForDisconnect. Now all tests are passing.

Reply:
I think a better/safer fix here would be to mack the mock return the default optional value, so that there is no timeout by default. Do you mind making that change?

Reply:
i will make that change. But the codec_client_test needs idletimeout set..will have to figure out how to set the value there

Reply:
ok. added MockIdleTimeEnabledClusterInfo which overrides that method and returns the value. This can be used in future for other tests if needed.

Reply:
i have reverted this. but added below. I can move it to the correct positions.